src/Common/src/Interop/OSX/Interop.libproc.cs

```diff
@@ -0,0 +1,542 @@
+            internal ulong          ri_cpu_time_qos_maintenance;
+            internal ulong          ri_cpu_time_qos_background;
+            internal ulong          ri_cpu_time_qos_utility;
+            internal ulong          ri_cpu_time_qos_legacy;
+            internal ulong          ri_cpu_time_qos_user_initiated;
+            internal ulong          ri_cpu_time_qos_user_interactive;
+            internal ulong          ri_billed_system_time;
+            internal ulong          ri_serviced_system_time;
+        }
+
+        // From proc_info.h
+        [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Ansi)]
+        internal unsafe struct proc_taskallinfo
+        {
+            internal proc_bsdinfo    pbsd;
+            internal proc_taskinfo   ptinfo;
+        }
+
+        // From proc_info.h
+        [StructLayout(LayoutKind.Sequential)]
+        internal unsafe struct proc_threadinfo
+        {
+            internal ulong      pth_user_time;
+            internal ulong      pth_system_time;
+            internal int        pth_cpu_usage;
+            internal int        pth_policy;
+            internal int        pth_run_state;
+            internal int        pth_flags;
+            internal int        pth_sleep_time;
+            internal int        pth_curpri;
+            internal int        pth_priority;
+            internal int        pth_maxpriority;
+            internal fixed byte pth_name[MAXTHREADNAMESIZE];
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct proc_fdinfo
+        {
+            internal int proc_fd;
+            internal uint proc_fdtype;
+        }
+
+        /// <summary>
+        /// Queries the OS for the PIDs for all running processes
+        /// </summary>
+        /// <param name="buffer">A pointer to the memory block where the PID array will start</param>
+        /// <param name="buffersize">The length of the block of memory allocated for the PID array</param>
+        /// <returns>Returns the number of elements (PIDs) in the buffer</returns>
+        [DllImport(Interop.Libraries.libproc, SetLastError = true)]
+        private static unsafe extern int proc_listallpids(
+            int*    pBuffer, 
+            int     buffersize);
+
+        /// <summary>
+        /// Queries the OS for the list of all running processes and returns the PID for each
+        /// </summary>
+        /// <returns>Returns a list of PIDs corresponding to all running processes</returns>
+        internal static unsafe int[] proc_listallpids()
+        {
+            // Get the number of processes currently running to know how much data to allocate
+            int numProcesses = proc_listallpids(null, 0);
+            if (numProcesses <= 0)
+            {
+                throw new System.ComponentModel.Win32Exception(Marshal.GetLastWin32Error(), SR.CantGetAllPids);
+            }
+
+            int[] processes = null;
```

Comment:
Nit: the initialization to null here shouldn't be necessary.  It's partly personal preference, but I try to avoid such things because then the compiler will help to flag places where I erroneously used with the variable without initializing it... by initializing it to null, the compiler will happily let you use it, resulting in more potential for null dereferencing.